lib/ui/src/modules/ui/components/left_panel/stories_tree/tree_decorators.js

```diff
@@ -34,41 +34,61 @@ ContainerDecorator.propTypes = {
   }).isRequired,
 };
 
-function HeaderDecorator(props) {
-  const { style, node } = props;
+function createHeaderDecoratorScope(parent) {
+  class HeaderDecorator extends React.Component {
+    constructor(...args) {
+      super(...args);
+      this.onKeyDown = this.onKeyDown.bind(this);
+    }
 
-  const newStyleTitle = {
-    ...style.title,
-  };
+    onKeyDown(event) {
+      const { onKeyDown } = parent;
+      const { node } = this.props;
+
+      onKeyDown(event, node);
+    }
+
+    render() {
+      const { style, node } = this.props;
+
+      const newStyleTitle = {
+        ...style.title,
+      };
 
-  const Icon = iconsMap[node.type];
+      const Icon = iconsMap[node.type];
 
-  if (!node.children || !node.children.length) {
-    newStyleTitle.fontSize = '13px';
+      if (!node.children || !node.children.length) {
+        newStyleTitle.fontSize = '13px';
+      }
+
+      return (
+        <div style={style.base} role="menuitem" tabIndex="0" onKeyDown={this.onKeyDown}>
+          {Icon && <Icon color={iconsColor} />}
+          <a style={newStyleTitle}>
+            {node.name}
+          </a>
+        </div>
+      );
+    }
   }
 
-  return (
-    <div style={style.base}>
-      {Icon && <Icon color={iconsColor} />}
-      <a style={newStyleTitle}>
-        {node.name}
-      </a>
-    </div>
-  );
-}
+  HeaderDecorator.propTypes = {
+    style: PropTypes.shape({
+      title: PropTypes.object,
```

Comment:
these things inside the shape should probably all be `.isRequired` from the looks of how they're used.